app/src/androidTest/java/org/dharmaseed/android/TalkRepositoryTest.java

```diff
@@ -0,0 +1,215 @@
+        SQLiteDatabase db = dbManager.getWritableDatabase();
+        InputStream fis = InstrumentationRegistry.getContext().getAssets().open("inserts.txt");
+        String line = "";
+        int ch;
+        // run every statement from "inserts.txt"
+        while ((ch = fis.read()) != -1)
+        {
+            if (ch < 8 || ch > 'z') continue;
+            line += (char) ch;
+            // execute the statement once we hit a semicolon followed by a newline
+            if (ch == ';' && fis.read() == '\n')
+            {
+                db.execSQL(line);
+                line = "";
+            }
+        }
+    }
+
+    @AfterClass
+    public static void tearDown() throws Exception
+    {
+        dbManager.close();
+    }
+
+    @Test
+    public void selectIds_isCorrect()
+    {
+        talkRepository = new TalkRepository(dbManager);
+
+        List<String> columns = new ArrayList<>();
+        columns.add(DBManager.C.Talk.TABLE_NAME + "." + DBManager.C.Talk.ID);
+        Cursor actualCursor = talkRepository.getTalks(columns, null, false, false);
+
+        SQLiteDatabase db = dbManager.getReadableDatabase();
+        // the getTalks() method should perform an equivalent query
+        Cursor expectedCursor = db.rawQuery("SELECT _id FROM talks ORDER BY rec_date DESC", null);
+
+        assertCursors(expectedCursor, actualCursor, columns);
+    }
+
+    @Test
+    public void getAllTalks_isCorrect()
+    {
+        talkRepository = new TalkRepository(dbManager);
+
+        Cursor actualCursor = talkRepository.getTalks(null, null, false, false);
+
+        SQLiteDatabase db = dbManager.getReadableDatabase();
+        // the getTalks() method should perform an equivalent query
+        Cursor expectedCursor = db.rawQuery("SELECT * FROM talks ORDER BY rec_date DESC", null);
+
+        List<String> columns = new ArrayList<>();
+
+        // compare every column
+        columns.add(DBManager.C.Talk.ID);
+        columns.add(DBManager.C.Talk.TITLE);
+        columns.add(DBManager.C.Talk.TEACHER_ID);
+        columns.add(DBManager.C.Talk.RECORDING_DATE);
+        columns.add(DBManager.C.Talk.VENUE_ID);
+        columns.add(DBManager.C.Talk.FILE_PATH);
+        columns.add(DBManager.C.Talk.AUDIO_URL);
+        columns.add(DBManager.C.Talk.UPDATE_DATE);
+        columns.add(DBManager.C.Talk.DESCRIPTION);
+        columns.add(DBManager.C.Talk.DURATION_IN_MINUTES);
+        columns.add(DBManager.C.Talk.RETREAT_ID);
+
+        assertCursors(expectedCursor, actualCursor, columns);
+    }
+
+    @Test
+    public void selectIdsTeacherIds_isCorrect()
+    {
+        talkRepository = new TalkRepository(dbManager);
+
+        List<String> columns = new ArrayList<>();
+        columns.add(DBManager.C.Talk.TABLE_NAME + "." + DBManager.C.Talk.ID);
+        columns.add(DBManager.C.Talk.TABLE_NAME + "." + DBManager.C.Talk.TEACHER_ID);
+        Cursor actual = talkRepository.getTalks(columns, null, false, false);
```

Comment:
Do you think it's worth adding a test or two to check that the starred and downloaded filters work as expected?

Reply:
(To be clear, I'm super happy we've gone from 0 tests to >0, an increase of infinity percent!  So I don't think we should hold off on merging this to add more if we want to do it later). :)

Reply:
For sure! I can add those now.

Reply:
:+1: Sounds good!